### ISSUE TITLE
Uncaught TypeError: Super expression must either be null or a function

### ISSUE THREAD

**Issue by triceras** [okta/okta-react#250]
### Describe the bug?

I am deploying okta-react for managing user authentication in my App. The app works fine locally but when I deployed the app in a Kubernetes cluster I got the following error.
```
Uncaught TypeError: Super expression must either be null or a function
```
The above error has been reported before and the solution proposed posted [here](https://stackoverflow.com/questions/68593219/okta-auth-js-uncaught-typeerror-super-expression-must-either-be-null-or-a-funct) did work for me as I was expecting. I can also confirm that the javascript added to `index.html` was executed.

My intention is to create a Login button on a Header so the user can log in and logout from the App.  I am new to ReactJS and I am learning.

How can I overcome this issue ??

My `App.js` is the following:
```
import { makeStyles, Paper, ThemeProvider } from '@material-ui/core';
import { createTheme } from '@material-ui/core/styles';
import React, { useEffect } from 'react';
import { BrowserRouter as Router, Route, Routes, useNavigate } from 'react-router-dom';
import { OktaAuth, toRelativeUrl } from '@okta/okta-auth-js';
import { Security, SecureRoute, LoginCallback } from '@okta/okta-react';


import Header from './components/Header';
import Create from './components/Pages/Create';
import Home from './components/Pages/Home';
import Update from './components/Pages/Update';

import config from './config';


    const theme = createTheme({
        palette: {
          primary: {
            main: "#3f51b5",
            light: '#3c44b126'
          },
          secondary: {
            main: "#f83245",
            light: '#f8324526'
          },
          background: {
            default: "#f4f5fd"
          },
        },
        overrides:{
          MuiAppBar:{
            root:{
              transform:'translateZ(0)'
            }
          }
        },
        props:{
          MuiIconButton:{
            disableRipple:true
          }
        },
        typography: {
            fontFamily: "Quicksand",
            fontWeightLight: 400,
            fontWeightRegular: 500,
            fontWeightMedium: 600,
            fontWeightBold: 700,
        }

    })

    const useStyles = makeStyles({

      pageContent: {
        margin: theme.spacing(3),
        padding: theme.spacing(2),
        width: '60%',
      },
    })

    const oktaAuth = new OktaAuth(config.oidc);

    const App = () => {

    const classes = useStyles();
    const [namespaces, setNamespaces] = React.useState([]);

    const navigate = useNavigate();

    const restoreOriginalUri = async (_oktaAuth, originalUri) => {
      navigate(toRelativeUrl(originalUri || '', window.location.origin), { replace: true });
    };


    useEffect(() => {
      fetch('https://secrets-interface-backend.cartax.dev/list_namespaces')
          .then(res => res.json())
          .then(data => {
              setNamespaces(data);
          })
          .catch(err => { console.log(err) })
          }, [])

    return (
        <ThemeProvider theme={theme}>
          <>
            <Security restoreOriginalUri={restoreOriginalUri} oktaAuth={oktaAuth}>
              <Header />
              <Routes>
                <Route path="/create" exact={true} element={<Paper className={classes.pageContent} elevation={2}> <Create namespaces={namespaces}/> </Paper>} />
                <Route path="/update" exact={true} element={<Paper className={classes.pageContent} elevation={2}> <Update namespaces={namespaces}/> </Paper>} />
                <Route path="/" exact={true} element={<Paper className={classes.pageContent} elevation={2}> <Home/> </Paper>} />
                <Route path="/login/callback" element={<LoginCallback />} />
              </Routes>

            </Security>
          </>
        </ThemeProvider>
    )
}

export default App;
```
The `Header.js` is the following:
```
import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import Menu from '@mui/material/Menu';
import MenuIcon from '@mui/icons-material/Menu';
import Container from '@mui/material/Container';
import Button from '@mui/material/Button';
import MenuItem from '@mui/material/MenuItem';
import KeyIcon from '@mui/icons-material/Key';
import { Link }  from 'react-router-dom';
import { useOktaAuth } from "@okta/okta-react";
import { OktaAuth, toRelativeUrl } from '@okta/okta-auth-js';

const pages = ['Create', 'Update'];
const title = 'Secrets Interface';

const Header = () => {
  const [anchorElNav, setAnchorElNav] = React.useState(null);

  const handleOpenNavMenu = (event) => {
    setAnchorElNav(event.currentTarget);
  };

  const handleCloseNavMenu = () => {
    setAnchorElNav(null);
  };

  const { authState, oktaAuth } = useOktaAuth();

  const loginWithRedirect = async () =>
    oktaAuth.signInWithRedirect({ originalUri: "/" });

  const logOut = async () => oktaAuth.signOut();

  if (!authState) {
    return <div>Loading...</div>;
  };


  return (
    <AppBar position="static">
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          <KeyIcon sx={{ display: { xs: 'none', md: 'flex' }, mr: 1 }} />
          <Typography
            variant="h6"
            noWrap
            component="a"
            href="/"
            sx={{
              mr: 2,
              display: { xs: 'none', md: 'flex' },
              fontFamily: 'Quicksand',
              fontWeight: 500,
              letterSpacing: '.3rem',
              color: 'inherit',
              textDecoration: 'none',
            }}
          >
            {title}
          </Typography>

          <Box sx={{ flexGrow: 1, display: { xs: 'flex', md: 'none' } }}>
            <IconButton
              size="large"
              aria-label="account of current user"
              aria-controls="menu-appbar"
              aria-haspopup="true"
              onClick={handleOpenNavMenu}
              color="inherit"
            >
              <MenuIcon />
            </IconButton>
            <Menu
              id="menu-appbar"
              anchorEl={anchorElNav}
              anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'left',
              }}
              keepMounted
              transformOrigin={{
                vertical: 'top',
                horizontal: 'left',
              }}
              open={Boolean(anchorElNav)}
              onClose={handleCloseNavMenu}
              sx={{
                display: { xs: 'block', md: 'none' },
              }}
            >
              {pages.map((page) => (
                <MenuItem key={page} onClick={handleCloseNavMenu}>
                  <Typography textAlign="center">
                    <Link style={{ textDecoration: "none", color: "gray" }} to={`/${page}`}>
                      {page}
                    </Link>
                  </Typography>
                </MenuItem>
              ))}
            </Menu>
          </Box>
          <KeyIcon sx={{ display: { xs: 'flex', md: 'none' }, mr: 1 }} />
          <Typography
            variant="h5"
            noWrap
            component="a"
            href=""
            sx={{
              mr: 2,
              display: { xs: 'flex', md: 'none' },
              flexGrow: 1,
              fontFamily: 'monospace',
              fontWeight: 700,
              letterSpacing: '.3rem',
              color: 'inherit',
              textDecoration: 'none',
            }}
          >
            {title}
          </Typography>
          <Box sx={{ flexGrow: 1, display: { xs: 'none', md: 'flex' } }}>
            {pages.map((page) => (
              <Button
                key={page}
                onClick={handleCloseNavMenu}
                sx={{ my: 2, color: 'white', display: 'block' }}
              >
                <Link style={{ textDecoration: "none", color: "white" }} to={`/${page}`}>
                      {page}
                </Link>
              </Button>
            ))}
          </Box>
          {
            authState.isAuthenticated ? (
                  <Button variant="contained" onClick={logOut} color="inherit">Logout</Button>
                ) : (
                  <Button variant="contained" onClick={loginWithRedirect} color="inherit">Login</Button>
                )
          }
        </Toolbar>
      </Container>
    </AppBar>
  );
};
export default Header;
```

### What is expected to happen?

The Application page should be able to render

### What is the actual behavior?

The Application page is completely blank and does not render on the browser. I got a single error message on the console log saying `2.8bfaec71.chunk.js:1 Uncaught TypeError: Super expression must either be null or a function`

![Screen Shot 2022-11-12 at 2 19 07 PM](https://user-images.githubusercontent.com/1896150/201496735-3b0fead4-b132-40ed-8c1a-f43490ef9668.png)

### Reproduction Steps?

I think it can be reproduced. Please let me know what information is required to reproduce the error.

### SDK Versions

```
"@okta/okta-auth-js": "^7.0.1",
"@okta/okta-react": "^6.7.0",
"@okta/okta-signin-widget": "^7.0.0",
```

### Execution Environment

The Application is a ReactJS frontend application that talks to a golang backend.

**Comment by denysoblohin-okta** [comment]
Could you please check solutions posted to the same problem and see if it works for you:
- https://devforum.okta.com/t/uncaught-typeerror-super-expression-must-either-be-null-or-a-function/15345/9
- https://stackoverflow.com/questions/30116430/reactjs-giving-error-uncaught-typeerror-super-expression-must-either-be-null-or?page=1&tab=votes#tab-top
Also can you debug the call stack of error and see the source of it?

**Comment by triceras** [comment]
Hi @denysoblohin-okta,

  Thanks for looking into this. All the links you provided have already been reviewed and their proposed solution implemented, however, those suggestions did not work in my case. I included the okta-auth-js CDN link in my index.html as suggested and it did not help me.  Here is the script I included.
```
 <script
      src="https://global.oktacdn.com/okta-auth-js/4.5.0/okta-auth-js.min.js"
      type="text/javascript"
    ></script>
    <script>
      var redirectUri = `${location.origin}/okta`;
        window.myOkta = new OktaAuth({
          issuer: "https://cartax.okta.com/oauth2/default",
          clientId: "0oa9rybfjfAcFpDbH4x7",
          redirectUri: window.location.origin + '/login/callback',
          scopes: ['openid', 'profile', 'email'],
          pkce: true,
          disableHttpsCheck: false,
          authParams: {
            issuer: "",
            responseType: ["id_token", "token"],
          },
        })
    </script>
```

We are not using class-based components but function-based components. I reviewed the package declarations on each react file and I could not identify any issues with Upper/Lower cases.

The call stack made to the React App is shown below. I got all responses back a 200 status code, but the App page fails to render.
![Screen Shot 2022-11-14 at 8 43 51 AM](https://user-images.githubusercontent.com/1896150/201731247-597adbb3-7c8e-4fe0-9987-508cfcbd9809.png).

Interestingly, when a call is made to the App URL I got the message `You need to enable JavaScript to run this app.`
![Screen Shot 2022-11-14 at 8 48 00 AM](https://user-images.githubusercontent.com/1896150/201731707-c8e2ffb5-1a2d-4984-816a-afe2c3c4fffc.png)

I tried to implement the suggestion made on the post [Solved - You Need to Enable Javascript to Run This App](https://www.upbeatcode.com/react/solved-you-need-to-enable-javascript-to-run-this-app/) but it did not help me.

**Comment by jaredperreault-okta** [comment]
What version of react are you using?

**Comment by triceras** [comment]
> What version of react are you using?

I am using `18.2.0`. It is also in the bug description.

**Comment by triceras** [comment]
Below is my index.js file.
```
import React from 'react'
import { createRoot } from 'react-dom/client';

import App from './App'
import './index.css'
import { BrowserRouter as Router } from 'react-router-dom'
import express from 'express'

const path = require('path');

createRoot(document.getElementById('root')).render(
    <Router>
        <App />
    </Router>
)

App.use(express.static(__dirname))

App.get("/*", (req, res) => {
    res.sendFile(path.join(__dirname, '/'));
});
```
And my package.json.
```
{
  "name": "frontend",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@emotion/react": "^11.10.4",
    "@emotion/styled": "^11.10.4",
    "@fortawesome/fontawesome-svg-core": "^6.2.0",
    "@fortawesome/free-solid-svg-icons": "^6.2.0",
    "@fortawesome/react-fontawesome": "^0.2.0",
    "@material-ui/core": "^4.12.4",
    "@material-ui/icons": "^4.11.3",
    "@mui/icons-material": "^5.10.9",
    "@mui/material": "^5.10.8",
    "@okta/okta-auth-js": "^7.0.1",
    "@okta/okta-react": "^6.7.0",
    "@okta/okta-signin-widget": "^7.0.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "bootstrap": "^5.2.2",
    "classmates": "^1.0.1",
    "express": "^4.18.2",
    "react": "^18.2.0",
    "react-bootstrap": "^2.5.0",
    "react-dom": "^18.2.0",
    "react-hook-form": "^7.37.0",
    "react-markdown": "^8.0.3",
    "react-router-dom": "^6.4.2",
    "react-scripts": "^2.1.3",
    "react-toastify": "^9.0.8",
    "remark-gfm": "^3.0.1",
    "semantic-ui-react": "^2.1.3",
    "uuid": "^9.0.0",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "react-scripts --openssl-legacy-provider start",
    "build": "react-scripts --openssl-legacy-provider build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "homepage": ".",
  "proxy": "http://localhost:3000"
}

```

**Comment by jaredperreault-okta** [comment]
I've made a ticket so we can review this internally

Internal Ref: OKTA-551970

**Comment by nzxwang** [comment]
Thanks for reporting and tracking this here. I experienced the same problem but resolved it by upgrading `react-scripts` from 3.0.1 to 5.0.1 as per [this suggestion](https://devforum.okta.com/t/uncaught-typeerror-super-expression-must-either-be-null-or-a-function/15345/9?u=nickw) in the first thread in https://github.com/okta/okta-react/issues/250#issuecomment-1313463715 posted by denysoblohin-okta
